web-client/public/js/update-app.js

```diff
@@ -14,7 +15,39 @@ const refreshApp = () => {
     }
 };
 
+const displayNetwork = (network, name) => {
+    uploadState.nodeColoring.reload(network, name);
+    if (document.getElementById("zoomSlider").disabled) {
+        document.getElementById("zoomSlider").disabled = false;
+    }
+
+    uploadState.currentNetwork = network;
+    console.log("Network: ", network); // Display the network in the console
+    $("#graph-metadata").html(network.genes.length + " nodes<br>" + network.links.length + " edges");
+
+    if (network.warnings.length > 0) {
+        displayWarnings(network.warnings);
+    }
+
+    $("#fileName").text(name); // Set the name of the file to display in the top bar
+    $("input[type='range']").off("input"); // I have no idea why I do this. Investigate later.
+
+    // If more things need to be turned off, we'll add them to this array
+    [ "#resetSliders", "#resetSlidersMenu", "#undoReset", "#undoResetMenu" ].forEach(
+        selector => $(selector).off("click")
+    );
+};
```

Comment:
@mihirsamdarshi This part may be of particular interest to you since you weren't sure how to handle `displayNetwork`. So note how it is now a helper function within _update-app_. I think that eventually, this might be integrated into `updateApp` itself. But for now we're keeping it separate since the original code only called this when a new network got loaded.